hw_25/tests/test_contacts.py

```diff
@@ -0,0 +1,171 @@
+"""Tests for full flow: signup-login-add_contact-update_contact-delete_contact"""
+
+
+import logging
+import pytest
+from selenium import webdriver
+from selenium.common import TimeoutException
+from selenium.webdriver.chrome.service import Service
+from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from webdriver_manager.chrome import ChromeDriverManager
+from hw_25.tests.test_data.constants import Urls, UserCredentials
+from hw_25.tests.test_data.test_data import TestDataUser, TestDataAddContact, TestDataUpdateContact
+from hw_25.resources.pages.login_page import LoginPage
+from hw_25.resources.pages.signup_page import SignupPage
+from hw_25.resources.pages.contactlist_page import ContactListPage
+from hw_25.resources.locators.logout_locators import LogoutLocators
+from hw_25.resources.locators.add_contact_locators import AddContactLocators
+from hw_25.resources.locators.update_contact_locators import UpdateContactLocators
+from hw_25.resources.locators.delete_contact_locators import DeleteContactLocators
+
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+@pytest.fixture(scope="module")
+def driver():
+    chrome_options = Options()
+    chrome_options.add_argument('--start-maximized')
+    service = Service(ChromeDriverManager().install())
+    driver = webdriver.Chrome(service=service, options=chrome_options)
+    driver.implicitly_wait(10)
```

Comment:
you have combined different approaches: use `implicitly_wait` or `WebDriverWait`

Reply:
fixed